hmkit-fleet/src/test/kotlin/com/highmobility/hmkitfleet/network/UtilityRequestsTest.kt

```diff
@@ -0,0 +1,194 @@
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package com.highmobility.hmkitfleet.network
+
+import com.highmobility.hmkitfleet.BaseTest
+import com.highmobility.hmkitfleet.model.Brand
+import com.highmobility.hmkitfleet.model.EligibilityStatus
+import com.highmobility.hmkitfleet.notExpiredAuthToken
+import io.mockk.coEvery
+import io.mockk.coVerify
+import io.mockk.mockk
+import kotlinx.coroutines.runBlocking
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.contentOrNull
+import kotlinx.serialization.json.jsonObject
+import kotlinx.serialization.json.jsonPrimitive
+import okhttp3.OkHttpClient
+import okhttp3.mockwebserver.MockResponse
+import okhttp3.mockwebserver.MockWebServer
+import okhttp3.mockwebserver.RecordedRequest
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.net.HttpURLConnection
+
+internal class UtilityRequestsTest : BaseTest() {
+    val mockWebServer = MockWebServer()
+    val client = OkHttpClient()
+    val authToken = notExpiredAuthToken()
+    val authTokenRequests = mockk<AuthTokenRequests>()
+
+    @BeforeEach
+    fun setUp() {
+        mockWebServer.start()
+        coEvery { authTokenRequests.getAuthToken() } returns Response(authToken)
+    }
+
+    @AfterEach
+    fun tearDown() {
+        mockWebServer.shutdown()
+    }
+
+    // get eligibility
+
+    @Test
+    fun getEligibilitySuccess() {
+        val mockResponse = MockResponse()
+            .setResponseCode(HttpURLConnection.HTTP_OK)
+            .setBody(
+                """
+                {
+                  "vin": "WBADT43452G296403",
+                  "eligible": true,
+                  "data_delivery": [
+                    "pull",
+                    "push"
+                  ]
+                }
+                """.trimIndent()
+            )
+        mockWebServer.enqueue(mockResponse)
+        val mockUrl = mockWebServer.url("").toString()
+        val webService = UtilityRequests(client, mockLogger, mockUrl, authTokenRequests)
+
+        val response = runBlocking {
+            webService.getEligibility("WBADT43452G296403", Brand.MERCEDES_BENZ)
```

Comment:
While the test is correct. `WBADT43452G296403` is a BMW vehicle. just to make the test semantically correct, I suggest to use bmw as brand